cli/js/lib.deno_runtime.d.ts

```diff
@@ -2130,6 +2130,80 @@ declare namespace Deno {
    */
   export const args: string[];
 
+  /** SignalStream represents the stream of signals, implements both
+   * AsyncIterator and PromiseLike */
+  export class SignalStream implements AsyncIterator<void>, PromiseLike<void> {
+    constructor(signal: typeof Deno.Signal);
+    then<T, S>(
+      f: (v: void) => T | Promise<T>,
+      g?: (v: void) => S | Promise<S>
+    ): Promise<T | S>;
+    next(): Promise<IteratorResult<void>>;
+    [Symbol.asyncIterator](): AsyncIterator<void>;
+    dispose(): void;
+  }
+  /**
+   * Returns the stream of the given signal number. You can use it as an async
+   * iterator.
+   *
+   *     for await (const _ of Deno.signal(Deno.Signal.SIGTERM)) {
```

Comment:
I'm not sure about the API Deno.Signal.SIGTERM... Would `Deno.signal("SIGTERM")` be better?

Reply:
@ry IMHO using enum is probably fine. This `Deno.Signal.*` is also shared with `kill()`

Reply:
ok - agreed